apps/site/src/_pages/docs/4_spec/02_graph-module.mdx

```diff
@@ -1,303 +1,445 @@
 # Graph Module
 
-This document (version 0.2) defines a **graph module** according to the terms set out in the [core specification](https://blockprotocol.org/docs/spec/core).
+This document (version 0.3) defines a **graph module** according to the terms set out in the [core specification](https://blockprotocol.org/docs/spec/core).
 
-The graph module describes how **entities** can be created, queried, updated, and linked together – including the **block entity**, represented by the **properties** sent to the block.
+The graph module describes how **ontology types** and **entities** can be created, queried, updated, and linked together – including the **block entity**, represented by the **properties** sent to the block.
 
 ## Glossary
 
-**Block entity**: an instance of data associated with the block, conforming to the block’s schema.
+**Entity:** an instance of data, often corresponding to a thing in the world, constrained by a particular entity type.
 
-**Entity:** an instance of data, often corresponding to a thing in the world, of a particular entity type.
+**Link Entity:** a kind of entity which represents a connection or relation between its left and right entity, constrained by a link entity type.
 
-**Entity type:** a specific type of entity, with a schema defining its structure. Entities have a type, and types have a schema.
+**Left Entity:** the entity which is on the "left" of a specific link entity, in most situations this can be thought of as the "source" of the link.
 
-**Graph:** a network of entities connected by links.
+**Right Entity:** the entity which is on the "right" of a specific link entity, in most situations this can be thought of as the "target" of the link.
 
-**Link:** a connection or relation between two entities.
+**Ontology Type:** an element of the Block Protocol Type System, represented as a self-contained schema accessible via a URI. Ontology types are composed together to describe the shape of entities within the graph.
 
-**Linked entities**: entities which are linked to the block entity, whether directly or via other entities.
+**Entity type:** an ontology type that defines the structure of an entity and its link entities. An entity type references other entity types and property types.
 
-**Properties:** the fields on a schema, what types of data are permitted as their values, and any other validation rules. Field types may be scalar (integer, string etc.), complex objects, collections (lists/arrays), or refer to another schema.
+**Link Entity type:** an entity type which has a special marker to say it constrains a link entity.
```

Comment:
```suggestion
**Link Entity type:** an entity type which has a special marker to say it defines a link entity.
```

Reply:
Do the types _define_ entities? They don't specify the properties of these things

Reply:
how about 'describes the shape of' or something? or 'constrains the shape / structure of'?

I just find 'constrains an x' to be less clear than it could